lib/dep-graph-builders/pnpm/lockfile-parser/lockfile-v6.ts

```diff
@@ -0,0 +1,84 @@
+import { PnpmWorkspaceArgs } from '../../types';
+import { PnpmDeps } from '../types';
+import { PnpmLockfileParser } from './lockfile-parser';
+import * as pathUtil from 'path';
+import { valid } from 'semver';
+
+export class LockfileV6Parser extends PnpmLockfileParser {
+  public settings;
+
+  public constructor(rawPnpmLock: any, workspaceArgs?: PnpmWorkspaceArgs) {
+    super(rawPnpmLock, workspaceArgs);
+    this.settings = rawPnpmLock.settings;
+  }
+
+  public parseDepPath(depPath: string) {
+    // Exclude transitive peer deps from depPath
+    // e.g. '/cdktf-cli@0.20.3(ink@3.2.0)(react@17.0.2)' -> cdktf-cli@0.20.3
+    depPath = this.excludeTransPeerDepsVersions(depPath);
+
+    // Check if path is absolute (doesn't start with '/')
+    // If it's not absolute, omit first '/'
+    depPath = LockfileV6Parser.isAbsolute(depPath)
+      ? depPath
+      : depPath.substring(1);
+
+    // Next, get version based on the last occurence of '@' separator
+    // e.g. @babel/code-frame@7.24.2 -> name: @babel/code-frame and version: 7.24.2
+    const sepIndex = depPath.lastIndexOf('@');
+    if (sepIndex === -1) {
+      return {};
+    }
+    const name = depPath.substring(0, sepIndex);
+    let version = depPath.substring(sepIndex + 1);
+    return {
+      name,
+      version,
+    };
+  }
+
+  public normalizeTopLevelDeps(
+    dependencies: Record<string, Record<string, string>>,
+    isDev: boolean,
+  ): PnpmDeps {
+    return Object.entries(dependencies).reduce(
+      (acc: PnpmDeps, [name, depInfo]) => {
+        const version = this.normalizeVersion(name, depInfo.version, isDev);
+        acc[name] = {
+          name,
+          version,
+          specifier: depInfo.specifier,
+          isDev,
+        };
+        return acc;
+      },
+      {},
+    );
+  }
+
+  public normalizePackagesDeps(
+    dependencies: Record<string, Record<string, string>>,
+    isDev: boolean,
+  ): Record<string, string> {
+    return Object.entries(dependencies).reduce(
+      (acc: Record<string, string>, [name, depInfo]) => {
+        const version = this.normalizeVersion(name, depInfo.version, isDev);
+        acc[name] = version;
+        return acc;
+      },
+      {},
+    );
+  }
+
+  public excludeTransPeerDepsVersions(fullVersionStr: string): string {
+    if (!fullVersionStr) {
+      return fullVersionStr;
+    }
+    const match = fullVersionStr.match(/([^)]*)\(/);
+    return match?.[1] ?? fullVersionStr;
+  }
+
+  public static isAbsolute(dependencyPath: string): boolean {
```

Comment:
Nit: rename the function to match the purpose ( `isAbsolute` is usually not being used for strings ) in this case, I would probably use `isAbsoluteDepenencyPath`